<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ImagePypelines</class>
 <widget class="QMainWindow" name="ImagePypelines">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1028</width>
    <height>620</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QWidget" name="graphicswidget" native="true">
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QToolBox" name="toolBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>200</width>
           <height>0</height>
          </size>
         </property>
         <property name="currentIndex">
          <number>1</number>
         </property>
         <widget class="QWidget" name="blockBox">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>500</width>
            <height>157</height>
           </rect>
          </property>
          <attribute name="label">
           <string>Builtin Blocks</string>
          </attribute>
         </widget>
         <widget class="QWidget" name="pipelineBox">
          <attribute name="label">
           <string>Builtin Pipelines</string>
          </attribute>
         </widget>
         <widget class="QWidget" name="backgroundBox">
          <attribute name="label">
           <string>Backgrounds</string>
          </attribute>
         </widget>
        </widget>
       </item>
       <item>
        <widget class="QFrame" name="frame">
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QComboBox" name="pipelineSelector"/>
          </item>
          <item>
           <widget class="QGraphicsView" name="graphicsView"/>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QtInterpreter" name="interpwidget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>200</height>
       </size>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1028</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionExit"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menuPipeline">
    <property name="title">
     <string>Pipeline</string>
    </property>
    <addaction name="actionNewPipeline"/>
    <addaction name="actionDelete_Pipeline"/>
    <addaction name="separator"/>
    <addaction name="actionRun"/>
   </widget>
   <widget class="QMenu" name="menuBlock">
    <property name="title">
     <string>Block</string>
    </property>
    <addaction name="actionMove_to_Front"/>
    <addaction name="actionSend_to_back"/>
    <addaction name="actionconnectBlocks"/>
    <addaction name="actionaddItem"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuPipeline"/>
   <addaction name="menuBlock"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNewPipeline"/>
   <addaction name="actionDelete_Pipeline"/>
   <addaction name="actionRun"/>
   <addaction name="actionDebug"/>
   <addaction name="separator"/>
   <addaction name="actionaddItem"/>
   <addaction name="actionDelete"/>
   <addaction name="actionselect"/>
   <addaction name="actionconnectBlocks"/>
   <addaction name="actionMove_to_Front"/>
   <addaction name="actionSend_to_back"/>
   <addaction name="separator"/>
  </widget>
  <action name="actionExit">
   <property name="icon">
    <iconset resource="diagramscene.qrc">
     <normaloff>:/images/exit.png</normaloff>:/images/exit.png</iconset>
   </property>
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="toolTip">
    <string>Close the application.</string>
   </property>
  </action>
  <action name="actionDelete">
   <property name="icon">
    <iconset resource="diagramscene.qrc">
     <normaloff>:/images/delete.png</normaloff>:/images/delete.png</iconset>
   </property>
   <property name="text">
    <string>Delete</string>
   </property>
   <property name="toolTip">
    <string>Delete the selected item(s)</string>
   </property>
  </action>
  <action name="actionconnectBlocks">
   <property name="icon">
    <iconset resource="diagramscene.qrc">
     <normaloff>:/images/linepointer.png</normaloff>:/images/linepointer.png</iconset>
   </property>
   <property name="text">
    <string>connectBlocks</string>
   </property>
   <property name="toolTip">
    <string>Connect two items</string>
   </property>
  </action>
  <action name="actionaddItem">
   <property name="icon">
    <iconset resource="diagramscene.qrc">
     <normaloff>:/images/addpipeline.png</normaloff>:/images/addpipeline.png</iconset>
   </property>
   <property name="text">
    <string>addItem</string>
   </property>
   <property name="toolTip">
    <string>Add a blank block</string>
   </property>
  </action>
  <action name="actionRun">
   <property name="icon">
    <iconset resource="diagramscene.qrc">
     <normaloff>:/images/run.png</normaloff>:/images/run.png</iconset>
   </property>
   <property name="text">
    <string>Run</string>
   </property>
   <property name="toolTip">
    <string>Run the active pipeline</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="actionselect">
   <property name="icon">
    <iconset resource="diagramscene.qrc">
     <normaloff>:/images/select.png</normaloff>:/images/select.png</iconset>
   </property>
   <property name="text">
    <string>select</string>
   </property>
   <property name="toolTip">
    <string>Select Mode</string>
   </property>
  </action>
  <action name="actionNewPipeline">
   <property name="icon">
    <iconset resource="diagramscene.qrc">
     <normaloff>:/images/addpipeline.png</normaloff>:/images/addpipeline.png</iconset>
   </property>
   <property name="text">
    <string>New Pipeline</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionMove_to_Front">
   <property name="icon">
    <iconset resource="diagramscene.qrc">
     <normaloff>:/images/bringtofront.png</normaloff>:/images/bringtofront.png</iconset>
   </property>
   <property name="text">
    <string>Move to Front</string>
   </property>
   <property name="toolTip">
    <string>Move item to the front</string>
   </property>
  </action>
  <action name="actionSend_to_back">
   <property name="icon">
    <iconset resource="diagramscene.qrc">
     <normaloff>:/images/sendtoback.png</normaloff>:/images/sendtoback.png</iconset>
   </property>
   <property name="text">
    <string>Send to back</string>
   </property>
   <property name="toolTip">
    <string>Send item to the back</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="icon">
    <iconset resource="diagramscene.qrc">
     <normaloff>:/images/about.png</normaloff>:/images/about.png</iconset>
   </property>
   <property name="text">
    <string>About</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+H</string>
   </property>
  </action>
  <action name="actionDelete_Pipeline">
   <property name="icon">
    <iconset resource="diagramscene.qrc">
     <normaloff>:/images/delete.png</normaloff>:/images/delete.png</iconset>
   </property>
   <property name="text">
    <string>Delete Pipeline</string>
   </property>
   <property name="toolTip">
    <string>Delete a Pipeline</string>
   </property>
  </action>
  <action name="actionDebug">
   <property name="icon">
    <iconset resource="diagramscene.qrc">
     <normaloff>:/images/bug.png</normaloff>:/images/bug.png</iconset>
   </property>
   <property name="text">
    <string>Debug</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QtInterpreter</class>
   <extends>QWidget</extends>
   <header>QtInterpreter</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="diagramscene.qrc"/>
 </resources>
 <connections/>
</ui>
